src/ocean/util/app/DaemonApp.d

```diff
@@ -323,6 +324,9 @@ public abstract class DaemonApp : Application,
     {
         super(name, desc);
 
+        // DaemonApp always handles SIGTERM:
+        settings.signals ~= core.sys.posix.signal.SIGTERM;
```

Comment:
What if SIGTERM is already in the list?

Reply:
Good point, I think either `SignalExt` or `DaemonApp` when instantiating should remove duplicates.

Reply:
I checked that, duplicates are ignored far down the call stack when registering in `SignalFD`.

Reply:
Maybe this should be a set instead (not for now, but for a future PR)

Reply:
Oh, wait, I just remembered nothing in ocean is using `SignalFD` anymore!

Reply:
Yeah, no deduplication is done: https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/util/app/ext/SignalExt.d#L409 Could you just use `uniq` either here or down in `SignalExt`?

Reply:
Updated:

```diff
diff --git a/src/ocean/util/app/DaemonApp.d b/src/ocean/util/app/DaemonApp.d
index 5db601e..4aa84d2 100644
--- a/src/ocean/util/app/DaemonApp.d
+++ b/src/ocean/util/app/DaemonApp.d
@@ -325,7 +325,9 @@ public abstract class DaemonApp : Application,
         super(name, desc);
 
         // DaemonApp always handles SIGTERM:
-        settings.signals ~= core.sys.posix.signal.SIGTERM;
+        auto term = core.sys.posix.signal.SIGTERM;
+        if (find(settings.signals, term) == settings.signals.length)
+            settings.signals ~= term;
```

Reply:
>Yeah, no deduplication is done: https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/util/app/ext/SignalExt.d#L409 Could you just use uniq either here or down in SignalExt?

Mihails said he checked and it's done further down:

>I checked that, duplicates are ignored far down the call stack when registering in SignalFD.

Reply:
I have assumed @nemanja-boric-sociomantic comment was that we no longer use `SignalFD` for it in most recent ocean. Was I wrong?

Reply:
Correct, we're no longer using `signalfd`, neither in `SignalExt`, nor in `EpollProcess`, since ocean v3.x.x.

Reply:
(we're using self-piping trick).